### PR TITLE
Add error builder to SVG widget

### PR DESCRIPTION
**Pull request by xal**
Related issues: #448, #535 
Similar old PR: https://github.com/dnfield/flutter_svg/pull/277

### CODE REVIEW

File: lib/svg.dart [hunk 1]
```diff
@@ -16,6 +16,13 @@ import 'src/svg/theme.dart';
 import 'src/utilities/file.dart';
 import 'src/vector_drawable.dart';
 
+/// Builder function to create an error widget.
+typedef LoadingErrorWidgetBuilder = Widget Function(
+    BuildContext context,
+    dynamic error,
```
Comment (dnfield) [inline lib/svg.dart:22]: Please avoid dynamic.

```suggestion
    Object error,
```

File: lib/svg.dart [hunk 2]
```diff
@@ -688,6 +701,9 @@ class SvgPicture extends StatefulWidget {
   /// The placeholder to use while fetching, decoding, and parsing the SVG data.
   final WidgetBuilder? placeholderBuilder;
 
+  /// The error widgeet to use while fetching, decoding, or parsing fails.
```
Comment (dnfield) [inline lib/svg.dart:704]:
```suggestion
  /// A builder to call if fetching, decoding, or parsing fails.
```

**Review by dnfield — CHANGES_REQUESTED** [review]
Please add a test.

**Comment by mekhri-codex** [comment]
@xal why it's closed? would it be posted later? I can't find a way to catch exceptions to display errorWidget

**Comment by xal** [comment]
> @xal why it's closed? would it be posted later? I can't find a way to catch exceptions to display errorWidget

Unfortunately I don't have time and enough experience with ui testing in flutter to write tests for this PR. I use my changes in my app via fork

**Pull request closed without merging**